### ISSUE TITLE
Tree View lacks typeahead functionality: Best Practice

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3603]
## Describe the bug

Tree Views lack Type-ahead functionality. Type-ahead is recommended for all trees, especially for trees with more than 7 root nodes. 

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Using the keyboard alone try to navigate any of the tree examples by typing the first character seen in any treenode.
- Observe that this is not possible

## Expected behavior

Implement Type-ahead functionality as described below.

- Type a character: focus moves to the next node with a name that starts with the typed character.
- Type multiple characters in rapid succession: focus moves to the next node with a name that starts with the string of characters typed.

See
https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by hippee-lee** [comment]
Is this search only for nodes in the DOM? Does it matter if they are hidden or visible?

**Comment by shravansofts** [comment]
This is an important feature.
Why this is marked as LowPriority? :(

**Comment by gnomeontherun** [comment]
It is low because its a recommendation but not a requirement, and it is also quite difficult to implement because of the nature of lazy loaded tree views not knowing the whole tree. We don't have a model of the data, but only know about the structure once it is rendered, so traversing to search for values is also a performance drain that we have to consider.

**Comment by chlane** [comment]
The WCAG verbiage says "Type-ahead is recommended for all trees, especially for trees with more than 7 root nodes:".  It looks like the use of the word "recommended" lowers its priority

**Comment by gnomeontherun** [comment]
Hi there, this is an automated message.

In an effort to dedupe and track accessibility reports consistently, this issue is being closed and tracked internally with our Accessibility team. It will still be triaged and addressed as part of ongoing work.
